src/widget/city_draw_highway.c

```diff
@@ -0,0 +1,57 @@
+#include "city_draw_highway.h"
+
+#include "assets/assets.h"
+#include "building/building.h"
+#include "city/view.h"
+#include "graphics/image.h"
+#include "map/aqueduct.h"
+#include "map/building.h"
+#include "map/grid.h"
+#include "map/random.h"
+#include "map/terrain.h"
+#include "map/tiles.h"
+
+static int highway_wall_direction_offsets[4] = { 1, -GRID_SIZE, -1, GRID_SIZE };
+
+static int is_highway_access(int grid_offset)
+{
+    if (map_terrain_is(grid_offset, TERRAIN_HIGHWAY | TERRAIN_ROAD | TERRAIN_GATEHOUSE | TERRAIN_ACCESS_RAMP)) {
+        return 1;
+    }
+    if (map_terrain_is(grid_offset, TERRAIN_BUILDING)) {
+        const building *b = building_get(map_building_at(grid_offset));
+        if (b->type == BUILDING_GRANARY) {
+            return grid_offset == b->grid_offset + map_grid_delta(1, 0) ||
+                grid_offset == b->grid_offset + map_grid_delta(0, 1) ||
+                grid_offset == b->grid_offset + map_grid_delta(2, 1) ||
+                grid_offset == b->grid_offset + map_grid_delta(1, 2);
+        }
+    }
+    return 0;
+}
+
+static void draw_wall_image(int grid_offset, int d, int x, int y, float scale)
+{
+    int direction_offset = grid_offset + highway_wall_direction_offsets[d];
+    if (is_highway_access(direction_offset)) {
+        return;
+    }
+    int wall_offset = (d + city_view_orientation() / 2) % 4;
+    int wall_image_id = assets_lookup_image_id(ASSET_HIGHWAY_BARRIER_START) + wall_offset;
+    image_draw_isometric_footprint_from_draw_tile(wall_image_id, x, y, 0, scale);
```

Comment:
I just have a question. Don't barriers also take a bit of the top too?

Reply:
Oh you mean like do they have a tiny top sprite? You know that’s a good question. They did when I made the placeholder images. I’ll have to look into that tonight. If they do I’ll probably need to get rid of my aqueduct image hack and draw out highway tops the same way I’m doing for footprints.

Reply:
I think they do, but I'm not entirely sure. I mean your new cut top pieces don't seem to have the top, but that may be a mistake in the xml as well.

It makes sense that they do have a top at least at the 2 topmost pixels, because the barrier is "higher" than the highway pavement.

Reply:
Doesn't look like there are any tops. I copied the XML over from what was there before, so it should all be about the same. Looks pretty good in screenshots too, I think:
![image](https://user-images.githubusercontent.com/2081059/200449850-fc396ca7-4a88-4d18-956f-924258d2807f.png)

Reply:
Just another random thought I had: it might be better that barriers are drawn as footprints instead of tops. If they were tops, they might draw on top of figures and we don't want that to happen.

Reply:
Yeah, it looks good. Ready to merge, then?

Reply:
Yep!